Feature/task3

HubertKarw opened the pull request.

HubertKarw committed:
java/product-project/task3/cart class

HubertKarw committed:
java/product-project/task3/cart tests

HubertKarw committed:
java/product-project/task4/Client class
added client class

HubertKarw committed:
java/product-project/task3/cart functionality
added adding and removing from cart

HubertKarw committed:
java/product-project/task3/formatting

HubertKarw committed:
Merge branch 'master' into feature/task3

# Conflicts:
#	src/com/company/Main.java
#	src/com/company/test/Test.java

HubertKarw committed:
java/product-project/task3/master merge changes

HubertKarw committed:
java/product-project/task3/decrease and increase stock when added/removed from cart

HubertKarw committed:
java/product-project/task3/formatting and client to string method
Added to string to client

HubertKarw committed:
java/product-project/task3/rename test
renaming test to meaningful name

HubertKarw committed:
java/product-project/task3/client test
added test for creating clint and main changes for client

HubertKarw committed:
java/product-project/task3/Changed functionality of sell product in ProductManager

kkpolak commented on src/com/company/Cart.java:
```diff
@@ -0,0 +1,60 @@
+            this.products.add(product);
+            product.decreaseStock(1);
+        } else {
+            throw new IllegalArgumentException("not enough in stock");
```
Taki message w exceptionie troche malo mowi, poprawniejszym podjesciem bylloby napisanie pelniejszej informacji typu: "cant add - product with id {id} has not enough..." - chodzi o to zeby dalo sie zidentyfikowac problem biznesowo

kkpolak commented on src/com/company/Cart.java:
```diff
@@ -0,0 +1,60 @@
+            this.products.remove(product);
+            product.increaseStock(1);
+        } else {
+            throw new IllegalArgumentException("no such product in cart");
```
ta sama uwaga co przy dodawnia

kkpolak commented on src/com/company/Cart.java:
```diff
@@ -0,0 +1,60 @@
+        this.client = client;
+    }
+
+    public void addToCart(Product product) {
```
ta logika jest dla mnie troche podejrzana, dodanie produktu do cartItems powoduje zmieniejszenie jego ilosci? Bardziej powinnp byc tak ze dodajemy go do cartItems i w liscie cardItems jego ilosc sie zwieksza, a w liscie produktow w magazanie ilosc sie zmniejsza, chyba ze czegos nie rozumiem

HubertKarw committed:
java/product-project/task3/cart functionality in product manager
added addtocart and remove from cart in product manager

HubertKarw committed:
java/product-project/task3//diferent implementation of addProductToCart

HubertKarw committed:
java/product-project/task3/clone and tests

HubertKarw committed:
java/product-project/task3/New add to cart and remove from cart in pm methods and test in main
updated product subclasses clone method and constructors
added functionality in cart and product manager

HubertKarw committed:
java/product-project/task3/live review and formatting changes
handling custom exception

HubertKarw committed:
java/product-project/task3/change in add and remove from cart in product manager
added isIDinCart method
replaced try catch statements with if statements

kkpolak commented on src/com/company/Cart.java:
```diff
@@ -0,0 +1,77 @@
+    }
+
+    public boolean isIdInCart(UUID id){
+        return this.getProducts().stream()
```
zerknij na metody ze stream-api anyMatch()

kkpolak commented on src/com/company/Main.java:
```diff
@@ -2,12 +2,41 @@
+//        Test.addingAndRemovingProductsToProductManager();
+//        Test.creatingClient();
+//        Test.creatingCartAndPrintingInformation();
+        Client client = new Client("123", "123");
```
test w osobnej metodzie - takie cos nie przejdzie na projekcie prawdziwym

kkpolak commented on src/com/company/ProductManager.java:
```diff
@@ -49,10 +53,61 @@ public Product findByID(UUID id) throws AttributeNotFoundException {
                 .orElseThrow(AttributeNotFoundException::new);
     }
 
+    public void addProductToCart(Cart cart, UUID productId, int quantity) {
```
formatowanie

HubertKarw committed:
java/product-project/task3/tests moved to test class, formatting, and changes to isidincart method
